packages/liveblocks-react-lexical/src/comments/threads-panel.tsx

```diff
@@ -0,0 +1,429 @@
+          top = positions.get(thread.id)!;
+        }
+
+        const isActive = activeThreads.includes(thread.id);
+
+        return (
+          <ThreadWrapper
+            key={thread.id}
+            Thread={Thread}
+            thread={thread}
+            onItemAdd={onItemAdd}
+            onItemRemove={onItemRemove}
+            style={{
+              position: "absolute",
+              transform: `translateX(${isActive ? -12 : 0}px) translateY(${top}px)`,
+              left: 0,
+              width: "100%",
+              paddingBottom: gap,
+              transition: "transform 0.2s",
+            }}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+interface ThreadWrapperProps extends ThreadProps {
+  Thread: ComponentType<ThreadProps>;
+}
+
+interface ThreadWrapperProps extends HTMLAttributes<HTMLDivElement> {
+  onItemAdd: (id: string, el: HTMLElement) => void;
+  onItemRemove: (id: string) => void;
+}
+
+function ThreadWrapper({
+  onItemAdd,
+  onItemRemove,
+  thread,
+  Thread,
+  ...divProps
+}: ThreadWrapperProps) {
+  const [editor] = useLexicalComposerContext();
+  const nodes = useThreadToNodes();
+  const divRef = useRef<HTMLDivElement>(null);
+
+  const activeThreads = useActiveThreads();
+
+  const isActive = activeThreads.includes(thread.id);
+
+  function handleThreadClick() {
+    const keys = nodes.get(thread.id);
+    if (keys === undefined || keys.size === 0) return;
+
+    if (activeThreads.includes(thread.id)) return;
+
+    editor.update(
+      () => {
+        const [key] = keys;
+        const node = $getNodeByKey(key);
+        if (!$isThreadMarkNode(node)) return;
+        node.selectStart();
+      },
+      {
+        onUpdate: () => {
+          const container = divRef.current;
+          if (container === null) return;
+
+          const composer = container.querySelector(".lb-composer-editor");
+          if (composer !== null) {
+            if (!(composer instanceof HTMLElement)) return;
+            composer.focus();
+          }
+          // If the composer is not found (and this can happen if `showComposer` is set to `false`), wait for the next render and try again to focus the composer.
+          else {
+            setTimeout(() => {
+              const composer = container.querySelector(".lb-composer-editor");
+              if (composer === null) return;
+              if (!(composer instanceof HTMLElement)) return;
+              composer.focus();
+            }, 0);
+          }
+        },
+      }
+    );
+  }
+
+  useLayoutEffect(() => {
+    const div = divRef.current;
+    if (div === null) return;
+
+    onItemAdd(thread.id, div);
+    return () => onItemRemove(thread.id);
+  }, [thread.id, onItemAdd, onItemRemove]);
+
+  return (
+    <div ref={divRef} {...divProps}>
+      <Thread
+        thread={thread}
+        data-state={isActive ? "active" : "inactive"}
+        onClick={handleThreadClick}
+        className={"lb-lexical-threads-panel-thread lb-elevation"}
```

Comment:
@marcbouchenoire I have added `lb-elevation` to threads inside the `ThreadsPanel` component (name not confirmed yet). This is what it looks like:

<img width="1116" alt="Screenshot 2024-07-08 at 9 18 59 PM" src="https://github.com/liveblocks/liveblocks/assets/35807215/885abc82-9fad-4ab3-a98d-182a41b65c71">

<img width="369" alt="Screenshot 2024-07-08 at 9 19 31 PM" src="https://github.com/liveblocks/liveblocks/assets/35807215/3c5ba08e-7443-4ef5-b6dd-5300a53bd876">

Do you think we should use a different elevation style? I am not sure the intention behind `lb-elevation` but was it for components that would float on top of something? These threads don't necessarily feel like floating elements. Would love your help figuring this out.

Reply:
I'll create a softer shadow and yep I don't think we should use `lb-elevation` here, only default to a shadow.